drivers/include/sx127x.h

```diff
@@ -0,0 +1,671 @@
+/*
+ * Copyright (C) 2016 Unwired Devices <info@unwds.com>
+ *               2017 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_sx127x SX127X
+ * @ingroup     drivers_netdev
+ * @brief       Semtech SX127X
+ *
+ * @{
+ *
+ * @file
+ * @brief       Public interface for SX127X driver
+ * @author      Eugene P. <ep@unwds.com>
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ */
+
+#ifndef SX127X_H
+#define SX127X_H
+
+#include "xtimer.h"
+#include "net/netdev.h"
+#include "periph/gpio.h"
+#include "periph/spi.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief SX127X LoRa configuration
+ * @{
+ */
+#define RF_FREQUENCY                                       (915000000UL)  /**< RF frequency in Hz, 868.9MHz */
+
+#define TX_OUTPUT_POWER                                    (10U)          /**< Radio power in dBm */
+
+#define LORA_PREAMBLE_LENGTH                               (8U)           /**< Preamble length, same for Tx and Rx */
+#define LORA_SYMBOL_TIMEOUT                                (10U)          /**< Symbols timeout (s) */
+
+#define LORA_FIX_LENGTH_PAYLOAD_ON                         false          /**< Set fixed payload length on */
+#define LORA_IQ_INVERSION                                  false          /**< Set inverted IQ on */
+/** @} */
+
+/**
+ * @brief SX127X device configuration
+ * @{
+ */
+#define SX127X_RADIO_WAKEUP_TIME                           (1000U)        /**< [us] */
+#define SX127X_RX_BUFFER_SIZE                              (256)          /**< RX buffer size */
+#define SX127X_RF_MID_BAND_THRESH                          (525000000UL)  /**< Mid-band threshold */
+#define SX127X_CHANNEL_HF                                  (868000000UL)  /**< HF channel */
```

Comment:
SX127x has some kind of differentiation for so-called High and Low-frequency bands. There is a two different matching network layouts are required for both of the bands and two different outputs for each of them are implemented in a device. So we have to differentiate Low (below 868 MHz) and High modes in software by currently used channel frequency to correctly setup device's registers and use correct output that is dependent on a current channel frequency.